src/tests/ftest/deployment/server_rank_failure.py

```diff
@@ -0,0 +1,423 @@
+
+        Returns:
+            bool: True if expected_health matches the one obtained from get-prop.
+
+        """
+        output = container.get_prop(properties=["status"])
+        actual_health = output["response"][0]["value"]
+
+        return actual_health == expected_health
+
+    def restart_all_servers(self):
+        """Restart daos_server on all nodes.
+
+        Stop on all nodes; both healthy nodes and the node where engines were killed.
+        Then start on all nodes.
+        """
+        self.log.info("Call systemctl stop daos_server")
+        # self.server_managers[0].stop() is the usual way, but this will also stop
+        # reset storage etc., which will make the server not start in the next step. It's
+        # also not the representation of the expected user behavior. Here, we just want
+        # to call systemctl stop daos_server.
+        # Calling manager.stop() is the equivalent of super().stop() in
+        # DaosServerManager.stop().
+        self.server_managers[0].manager.stop()
+        self.log.info("Start daos_server and detect the DAOS I/O engine message")
+        self.server_managers[0].restart(hosts=self.hostlist_servers)
+
+    def kill_engine(self, engine_kill_host):
+        """Kill engine on the given host.
+
+        Args:
+            engine_kill_host (str): Hostname to kill engine.
+        """
+        pattern = self.server_managers[0].manager.job.command_regex
+        result = stop_processes(hosts=[engine_kill_host], pattern=pattern)
+        if 0 in result and len(result) == 1:
+            self.log.info("No remote daos_engine process killed!")
+        else:
+            self.log.info("daos_engine in %s killed", engine_kill_host)
+
+    def verify_ior_worked(self, ior_results, job_num, errors):
+        """Verify that the IOR worked.
+
+        Args:
+            ior_results (dict): Dictionary that contains the IOR run results.
+            job_num (int): Job number used for the IOR run.
+            errors (list): Error list used in the test.
+        """
+        self.log.info(ior_results)
+        if not ior_results[job_num][0]:
+            ior_error = ior_results[job_num][1]
+            errors.append("Error found in IOR job {}! {}".format(job_num, ior_error))
+
+    def verify_rank_failure(self, ior_namespace, container_namespace):
+        """Verify engine failure can be recovered by restarting daos_server.
+
+        1. Create a pool and a container. Create a container with or without redundancy
+        factor based on container_namespace.
+        2. Get the amount of IO inflow to each rank. We expect them to be 0.
+        3. Run IOR with given object class (IO experiment run).
+        4. Get the amount of IO inflow again.
+        5. Get the rank with the largest amount of inflow. Find out its hostname. We'll
+        kill engines here. If we use replication, some ranks receive more data than
+        others. e.g., With RP_2G1, two ranks get most of the data.
```

Comment:
May be we can include this section of identifying where the data will go for the main ior run.
This can be done by increasing the stonewalling value for main ior run to something like 300 secs, and then perform these same steps above for the main ior command and finally stop the ranks you identify.

Reply:
Latest config uses `*X` object classes so all targets get data